app/services/container_dashboard_service.rb

```diff
@@ -0,0 +1,82 @@
+class ContainerDashboardService
+  def initialize(provider_id)
+    @provider_id = provider_id
+    @ems = ManageIQ::Providers::ContainerManager.find(@provider_id) unless @provider_id.nil?
+  end
+
+  def all_data
+    {
+      :status    => status,
+      :providers => providers
+    }
+  end
+
+  def status
+    if @ems.present? && @ems.kind_of?(ManageIQ::Providers::Openshift::ContainerManager)
+      routes_count = @ems.container_routes.count
+    else
+      routes_count = ContainerRoute.count
+    end
+
+    {
+      :nodes      => {
+        :count        => @ems.present? ? @ems.container_nodes.count : ContainerNode.count,
+        :errorCount   => 0,
+        :warningCount => 0
+      },
+      :containers => {
+        :count        => @ems.present? ? @ems.containers.count : Container.count,
+        :errorCount   => 0,
```

Comment:
is error count needed? afaik it's still not used in the cards in the current implementation